src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -150,13 +150,17 @@ def __init__(
         table_name: Iterable[str],
         column_names: Optional[List[str]],
         mode: SaveMode,
-        query: Optional[LogicalPlan],
+        query: LogicalPlan,
         table_type: str = "",
         clustering_exprs: Optional[Iterable[Expression]] = None,
         comment: Optional[str] = None,
         is_generated: bool = False,
     ) -> None:
         super().__init__()
+
+        assert (
```

Comment:
same here, I'll avoid assertion

Reply:
After the refactoring, the SnowflakeCreateTable always attached with a df._plan, and we rely on this fact to make sure we can perform optimization on the child, i want to make sure such assumption is not broken in the future.